modules/navier_stokes/src/userobjects/INSFVRhieChowInterpolator.C

```diff
@@ -219,6 +228,9 @@ INSFVRhieChowInterpolator::fillARead()
 
     _a_data_provided = true;
     _a_aux.resize(libMesh::n_threads());
+    if (getParam<bool>("force_preaux"))
+      mooseWarning("Executing the Rhie Chow user object before auxkernels is only valid if the "
+                   "auxvariables involved are constant over linear iterations");
```

Comment:
It seems like we will emit warnings when using the action even when the user themselves have not requested any non-standard behavior. This can train them to not respect warnings ... but at the same time it is good to tell them where possible issues might arise. 

Yucko. This is a really good example of wanting the arbitrary dep resolution. Just having it for user objects alone would fix this problem.

Reply:
true true. Maybe that is the smaller scope to carve out on this

Reply:
I've created #22255 which I don't think should be very hard to implement

Reply:
ok so now we are hitting errors with the PINSFV + reconstruction because the initial setup on a parsed function is not being run before preaux I think ...
lovely haha

Reply:
pre-aux user object `execute` should be run after functions have their `initialSetup` done

Reply:
This would make sense but smh getting a null ptr at evaluate time. 
I can investigate. 

Is the other modification OK ? (the flag on area PP)